core/primitives/src/profile_data_v3.rs

```diff
@@ -113,39 +124,44 @@ impl ProfileDataV3 {
                 // handle this case, we would need to explicitly count number of calls, not just
                 // the total gas usage.
                 // We don't have such costs at the moment, so this case is not implemented.
-                debug_assert!(key.gas(ext_costs_config) > 0 || key.compute(ext_costs_config) == 0);
+                debug_assert!(
+                    key.gas(ext_costs_config) > Gas::from_gas(0)
+                        || key.compute(ext_costs_config) == 0
+                );
 
-                if *value == 0 {
-                    return *value;
+                if *value == Gas::from_gas(0) {
+                    return (*value).as_gas();
                 }
                 // If the `value` is non-zero, the gas cost also must be non-zero.
-                debug_assert!(key.gas(ext_costs_config) != 0);
-                ((*value as u128).saturating_mul(key.compute(ext_costs_config) as u128)
-                    / (key.gas(ext_costs_config) as u128)) as u64
+                debug_assert!(key.gas(ext_costs_config) != Gas::from_gas(0));
+                (((*value).as_gas() as u128).saturating_mul(key.compute(ext_costs_config) as u128)
```

Comment:
grep for "as u128" in your [PR diff](https://github.com/near/nearcore/pull/13964.patch) and try to replace all of the occurrences with `.into()`